### ISSUE TITLE
Use addressfield as a form element type

### ISSUE THREAD

**Issue by alanmels** [backdrop-contrib/addressfield#35]
Would be wonderful to implement what's been suggested and discussed on https://www.drupal.org/project/addressfield/issues/970048 and would be able to use the `addressfield` type form elements in custom modules, something like:

```php
$form['personalinfo']['address'] = array(
    '#type' => 'addressfield',
    '#title' => t('Address'),
    '#required' => TRUE,
    '#default_value' => array('country' => 'Denmark'),
    '#description' => "Please enter your address.",
    '#default_widget' => 'addressfield_standard',
    '#default_formatter' => 'addressfield_default',
  );

```

The thread contains several patches and as https://www.drupal.org/project/addressfield/issues/3017567 shows there are Drupal 7 projects successfully using the patched version.

**Comment by argiepiano** [comment]
Thanks @alanmels  - this is an interesting proposal. I see this has been open in the Drupal 7 queue for a long time and is postponed, even though there have been a few releases in between. I need to read through the thread, and will get back to discuss.

**Comment by alanmels** [comment]
@argiepiano, thanks for taking a look. I have a working version with the PR on https://github.com/backdrop-contrib/addressfield/pull/36 applied. 

It's ok if it doesn't go in as we will keep using the patched version for a customer project that really needed the address field implemented not on a Backdrop entity, but on a custom module.

**Comment by jenlampton** [comment]
This sounds like a good idea, but the PR has some merge conflicts after the recent changes to addressfield. I also ned to review the drupal issue to see why the feature was postponed.

**Comment by argiepiano** [comment]
@alanmels, I'm looking at the Drupal queue - there are a multitude of patches offered by different people, many of them with different problems. It'd be helpful to know which patch did you base yours on? Have you had a chance to test the different issues people in that queue had with the patches?

Also, your PR has conflicts that need to be resolved. Can you please resolve them to allow for testing?

**Comment by alanmels** [comment]
Oh, @argiepiano, if only I remembered (it can be of course tracked by commit history) as we moved on since long. Unfortunately, I don't have time to make any correction at the moment. All I can say that it's working great for us as we can now create addressfields in custom code by simply declaring the following array:

```php
$form['personalinfo']['address'] = array(
    '#type' => 'addressfield',
    '#title' => t('Address'),
    '#required' => TRUE,
    '#default_value' => array('country' => 'Denmark'),
    '#description' => "Please enter your address.",
    '#default_widget' => 'addressfield_standard',
    '#default_formatter' => 'addressfield_default',
  );
```

Sorry for not being helpful so much, but time constrains dictate.

**Comment by argiepiano** [comment]
@alanmels I understand, but please realize that we can't merge a PR with conflicts. So this will need to stay as a PR, and hopefully people who need it will have the skills to adapt it to make it work with the latest version of Addressfield. If you ever get the time, please consider coming back to this. I think it's a useful feature.

**Comment by alanmels** [comment]
Hi @argiepiano,

Thanks a lot for your comment! 😊 Just to clarify — when this PR was first opened, there weren’t any conflicts and it was ready to merge. Over time, as other changes went in, conflicts naturally appeared.

I totally understand how things get busy (same on my side — time has been really scarce), which is why it’s always so helpful when PRs get reviewed and merged a bit sooner. We’re actually starting another project now that would really benefit from the feature in this PR, which is what brought me back here.

I won’t be able to resolve the conflicts myself right now, but I’d be so happy if someone else could jump in and help update it. I still believe this feature would be super useful!

**Comment by argiepiano** [comment]
@alanmels since we are all busy, and the original PRs from Drupal were never applied as they had different problems, my suggestion is that this feature be made into a separate module that doesn't have the potential to mess up addressfield. If I was the only maintainer of addressfield I could perhaps be more helpful.

**Comment by argiepiano** [comment]
BTW, that's what I ended up doing with the addressfield webform component - the review process by my co-maintainers never happened, and I felt it was faster and more helpful to split the feature into a separate module.

**Comment by alanmels** [comment]
@argiepiano fair enough 👍 — following your suggestion, we went ahead and split it out. Here’s the separate module: https://github.com/backdrop-contrib/address

**Comment by argiepiano** [comment]
Wonderful! Thanks so much
